src/inspector_agent.cc

```diff
@@ -438,6 +442,7 @@ void AgentImpl::OnRemoteDataIO(inspector_socket_t* socket,
     if (client_socket_ == socket) {
       String16 message(TAG_DISCONNECT, sizeof(TAG_DISCONNECT) - 1);
       client_socket_ = nullptr;
+      disconnecting_ = true;
```

Comment:
This seems prone to race condition. Variable is read and modified on 2 threads without any synchronization.

I wonder if it would be feasable to ::DispatchMessages when entering the main loop.